File: 125. Valid Palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int left = 0, right = s.length() - 1;

        while(left < right){
            char cLeft = s.charAt(left);    
            char cRight = s.charAt(right);

            if(!Character.isAlphabetic(cLeft) && !Character.isDigit(cLeft)) ++left;
            else if(!Character.isAlphabetic(cRight) && !Character.isDigit(cRight)) --right;
            else if(Character.compare(Character.toLowerCase(cLeft), Character.toLowerCase(cRight)) == 0) {
                ++left;
                --right;
            } else{
                return false;
            }
        }

        return true;
    }
}

File: 125. Valid Palindrome.cpp
#include<iostream>
#include<string.h>
using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        int left = 0, right = s.size() - 1;

        while(left < right){

            if(!isalpha(s[left]) && !isdigit(s[left])) ++left;
            else if(!isalpha(s[right]) && !isdigit(s[right])) --right;
            else if(tolower(s[left]) == tolower(s[right])){
                ++left;
                --right;
            }
            else return false;
        }
         
        return true;
    }
};